cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/golang/GoUtils.kt

```diff
@@ -212,8 +218,27 @@ internal class Project {
 
             log.debug("GOROOT/src is located @ {}", stdLib)
 
-            files += deps.flatMap { gatherGoFiles(stdLib.resolve(it), false) }
-            files += gatherGoFiles(topLevel)
+            // Build directories out of deps
+            val dirs: List<File> =
+                deps.mapNotNull {
+                    if (!it.contains(".")) {
+                        // without a dot, it is a stdlib package
+                        stdLib.resolve(it)
+                    } else if (it.startsWith("vendor")) {
+                        // if it already starts with vendor, it is a vendored stdlib package, we
+                        // don't really include these for now since they blow up the stdlib
+                        null
+                    } else if (it.startsWith(module.module.mod.path)) {
+                        topLevel.resolve(it.substringAfter(module.module.mod.path))
+                    } else {
+                        // for all other dependencies, we try whether they are vendored
+                        topLevel.resolve("vendor").resolve(it)
```

Comment:
I wpuld want a more detailed explanation what differentiates this from the `it.startsWith("vendor")`